lib/Sema/CSApply.cpp

```diff
@@ -1284,7 +1284,11 @@ namespace {
         } else {
           auto keyPathBGT = keyPathTy->castTo<BoundGenericType>();
           baseTy = keyPathBGT->getGenericArgs()[0];
-          
+
+          // Coerce the index to the key path's type
+          if (!keyPathTy->isEqual(keyPathExprTy))
```

Comment:
Do we need this test? There's an `isEqual()` call near the top of `coerceToType()` that might be equivalent.

Reply:
It is equivalent, and we don't need the test. I was following along with the similar code just below this that tests, but there it's needed because of the `getRValueType`. Not needed here.

Thanks!